### PR TITLE
sample projects for java

### PR DESCRIPTION
**Pull request by andxu**

### CODE REVIEW

File: samples/java/getting-started/mywebapi/src/main/java/com/ms/sample/mywebapi/Greeting.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package com.ms.sample.mywebapi;
+
+public class Greeting {
```
Comment (stepro) [inline samples/java/getting-started/mywebapi/src/main/java/com/ms/sample/mywebapi/Greeting.java]: I don't think we need this in the sample.

File: samples/java/getting-started/mywebapi/src/main/java/com/ms/sample/mywebapi/GreetingController.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package com.ms.sample.mywebapi;
+
+import java.util.concurrent.atomic.AtomicLong;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class GreetingController {
```
Comment (stepro) [inline samples/java/getting-started/mywebapi/src/main/java/com/ms/sample/mywebapi/GreetingController.java]: Don't need this.

File: samples/java/getting-started/mywebapi/src/main/java/com/ms/sample/mywebapi/HomeController.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+@RestController
+public class HomeController {
+
+    @RequestMapping("/")
+    public String index() {
+        return "Hello from Spring Boot!";
+    }
+
+    @RequestMapping("/version")
```
Comment (stepro) [inline samples/java/getting-started/mywebapi/src/main/java/com/ms/sample/mywebapi/HomeController.java]: Not needed.

File: samples/java/getting-started/mywebapi/src/test/java/com/ms/sample/mywebapi/DemoApplicationTests.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
+package com.ms.sample.mywebapi;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+public class DemoApplicationTests {
```
Comment (stepro) [inline samples/java/getting-started/mywebapi/src/test/java/com/ms/sample/mywebapi/DemoApplicationTests.java]: Is this file necessary?

File: samples/java/getting-started/webfrontend/src/main/resources/static/index.html [hunk 5]
```diff
@@ -0,0 +1,10 @@
+<!DOCTYPE HTML>
```
Comment (stepro) [inline samples/java/getting-started/webfrontend/src/main/resources/static/index.html]: What do you think about using the same static content as the Node.js app so there is a little more substance to the webfrontend (it uses Angular in a very minimal fashion).

File: samples/java/getting-started/webfrontend/src/test/java/com/ms/sample/SampleTest.java [hunk 6]
```diff
@@ -0,0 +1,14 @@
+package com.ms.sample;
+
+import static org.junit.Assert.assertEquals;
+import org.junit.Test;
+
+public class SampleTest {
```
Comment (stepro) [inline samples/java/getting-started/webfrontend/src/test/java/com/ms/sample/SampleTest.java]: Don't need this I think.

File: samples/java/getting-started/simpleapp/.vscode/launch.json [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
+    "version": "0.2.0",
+    "configurations": [
+        {
+            "type": "javaremote",
+            "name": "Attach To Remote Agent (AZDS)",
+            "request": "attach",
+            "hostName": "localhost",
+            "preLaunchTask": "azds: prelaunch",
+            "port": 53456
```
Comment (stepro) [inline samples/java/getting-started/simpleapp/.vscode/launch.json]: Please use a different random port.

File: samples/java/getting-started/simpleapp/src/main/java/com/ms/samples/App.java [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+
+/**
+ * Hello world!
+ *
+ */
+public class App 
+{
+    public static void main( String[] args )
+    {
+        System.out.println( "Hello World!");
```
Comment (stepro) [inline samples/java/getting-started/simpleapp/src/main/java/com/ms/samples/App.java]: This app is not a web server and doesn't run as a service. It isn't very useful to have this as an example in dev spaces.

File: samples/java/getting-started/webfrontend/.vscode/tasks.json [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+            "label": "azds: prelaunch",
+            "command": "azds",
+            "args": [
+                "up",
+                "--port=57583:7000",
+                "--detach",
+                "--keep-alive",
+                "--",
+                "/javadebug/start.sh",
+                "mvn",
```
Comment (stepro) [inline samples/java/getting-started/webfrontend/.vscode/tasks.json]: Does this launch the Java app in the same manner as before? Waiting for a debugger to attach on port 8000?

File: samples/java/getting-started/webfrontend/pom.xml [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.springframework.boot</groupId>
+                <artifactId>spring-boot-maven-plugin</artifactId>
+                <version>2.0.3.RELEASE</version>
+                <configuration>
+                    <addResources>true</addResources>
+                    <jvmArguments>
+                        -agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=8000
```
Comment (stepro) [inline samples/java/getting-started/webfrontend/pom.xml]: Ah, I see. It's a little odd to include debug specific information in pom.xml vs. in the launch specific task in VS Code (which is the only place it is actually used). What does spring-boot:run give us?

Comment (andxu) [reply samples/java/getting-started/webfrontend/pom.xml]: we can use the MAVEN_DEBUG_OPTS env variable:
like: "agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=8000    "

File: samples/java/getting-started/webfrontend/.vscode/tasks.json (hunk 9 above)
Comment (andxu) [reply samples/java/getting-started/webfrontend/.vscode/tasks.json]: see reply in pom.xml

File: samples/java/getting-started/webfrontend/pom.xml (hunk 10 above)
Comment (stepro) [reply samples/java/getting-started/webfrontend/pom.xml]: I don't think dev spaces supports yet passing custom environment variables when starting the container. I think I remember spring-boot:run is what allows us to dynamically update html inside the container and have the app pick it up?
